lib/delayed/priority.rb

```diff
@@ -147,6 +173,14 @@ def <=>(other)
       to_i <=> other
     end
 
+    def -(other)
+      to_i - other.to_i
+    end
+
+    def +(other)
+      to_i + other.to_i
+    end
```

Comment:
I was thinking about what it meant to always call `to_i` on other, and also what it means to always respond with an integer. For example:

```
Priority.new(1) + Priority.new(10)
```

I would expect that to return `Priority.new(11)`, not `11`.

And also these examples:

```ruby
Priority.new(10) + 1
```

Should also probably return `Priority.new(11)`.

But I'm not sure what this should return:

```ruby
Priority.new(10) + 0.9
```
(Should it round to `Priority.new(11)`, or raise an error?)

Maybe we don't need to handle that last case for now.

So I think that leaves us with this:

```suggestion
    def -(other)
      new(to_i - other.to_i)
    end

    def +(other)
      new(to_i + other.to_i)
    end
```

Though we could do what we do in `<=>` and cast other to_i only if it's a priority:

```ruby
other = other.to_i if other.is_a?(self.class)
```

WDYT?

Reply:
Oh yeah that's a good call! We should return a Priority instance, since that is the caller.

Hmm yeah the math complexities are interesting. I would say we probably kick the can on dealing with floats? Right now the `initialize` method (and the mid_point method) will ensure all instances have integer values.